transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -136,33 +135,27 @@ private ChannelPipeline addFirst(
             newCtx = new DefaultChannelHandlerContext(this, invoker, filterName(name, handler), handler);
             executor = executorSafe(invoker);
 
+            addFirst0(newCtx);
```

Comment:
consider inlineing code ... function is only called here and it will clarify synchronization assumptions in `addFirst0`